### PR TITLE
Parameterizable Frequency and Vivado Strategy Selection

### PR DESCRIPTION
**Pull request by davidbiancolin**
This works by synthesizing the PLL based on an TCL variable we emit from the generator, instead of selecting from a set of 6.

### CODE REVIEW

**Comment by sagark** [comment]
@davidbiancolin is this ready to review/merge?

**Comment by davidbiancolin** [comment]
Pending resolution of the SuperNode timing violations. I can't see how this caused them.

**Comment by davidbiancolin** [comment]
@sagark I propose we move to defining frequencies on a per-target basis as demostrated in the SKETCH commit. If we don't like it i'll just revert that change.

File: deploy/sample-backup-configs/sample_config_build_recipes.ini [hunk 1]
```diff
@@ -36,14 +36,14 @@ deploytriplet=None
...
-PLATFORM_CONFIG=FireSimDDR3FRFCFSLLC4MBConfig90MHz
+PLATFORM_CONFIG=F90MHz_FireSimDDR3FRFCFSLLC4MBConfig
 instancetype=c4.4xlarge
 deploytriplet=None
 
 [firesim-quadcore-no-nic-ddr3-llc4mb]
 DESIGN=FireSimNoNIC
 TARGET_CONFIG=FireSimRocketChipQuadCoreConfig
-PLATFORM_CONFIG=FireSimDDR3FRFCFSLLC4MBConfig90MHz
+PLATFORM_CONFIG=F90MHz_ireSimDDR3FRFCFSLLC4MBConfig
```
Comment (sagark) [inline deploy/sample-backup-configs/sample_config_build_recipes.ini]: the second part of this is broken

Comment (davidbiancolin) [reply deploy/sample-backup-configs/sample_config_build_recipes.ini]: Will fix if we're down to make this change.

**Comment by davidbiancolin** [comment]
@sagark This is ready for re-review.

**Comment by sagark** [comment]
oh i forgot, can you update docs for this?

**Comment by davidbiancolin** [comment]
> oh i forgot, can you update docs for this?

Done!

**Pull request merged**